src/qmeta/QMetaMysql.cc

```diff
@@ -794,11 +797,59 @@ void QMetaMysql::saveResultQuery(QueryId queryId, std::string const& query) {
         throw QueryIdError(ERR_LOC, queryId);
     } else if (results.getAffectedRows() > 1) {
         throw ConsistencyError(ERR_LOC, "More than one row updated for query ID " +
-                                                boost::lexical_cast<std::string>(queryId) + ": " +
-                                                boost::lexical_cast<std::string>(results.getAffectedRows()));
+                                                boost::lexical_cast<string>(queryId) + ": " +
+                                                boost::lexical_cast<string>(results.getAffectedRows()));
     }
 
     trans->commit();
 }
 
+void QMetaMysql::addQueryMessages(QueryId queryId, shared_ptr<qdisp::MessageStore> const& msgStore) {
+    int msgCount = msgStore->messageCount();
+    // &&& a bit of code duplication between this and MessageTable::_saveQueryMessages
+    for (int i = 0; i != msgCount; ++i) {
+         qdisp::QueryMessage const& qMsg = msgStore->getMessage(i);
+
+         try {
+             _addQueryMessage(queryId, qMsg);
+         } catch (qmeta::SqlError const& ex) {
+             LOGS(_log, LOG_LVL_ERROR, "UserQuerySelect::_qMetaUpdateMessages failed " << ex.what());
+         }
+    }
+}
+
+void QMetaMysql::_addQueryMessage(QueryId queryId, qdisp::QueryMessage const& qMsg) {
+    lock_guard<mutex> sync(_dbMutex);
+
+    auto trans = QMetaTransaction::create(*_conn);
+
+    // build query
+    std::string severity = (qMsg.severity == MSG_INFO ? "INFO" : "ERROR");
+
+    string query =
+            "INSERT INTO QMessages (queryId, chunkId, code, message, severity, timestamp) VALUES (";
+    query += boost::lexical_cast<string>(queryId);
+    query += ", ";
+    query += boost::lexical_cast<string>(qMsg.chunkId);
+    query += ", ";
+    query += _conn->escapeString(severity);
+    query += ", ";
+    query += _conn->escapeString(qMsg.description);
+    query += ", ";
+    query += boost::lexical_cast<string>(qMsg.timestamp);
+    query += ")";
```

Comment:
This "staircase" could be written in one (or two, if needed to wrap after 110 characters) line:
```
string const query = "INSERT INTO QMessages (queryId, chunkId, code, message, severity, timestamp) VALUES (" + to_string(queryId) +  ", "to_string(qMsg.chunkId) + ", "
    + _conn->escapeString(severity) + ", " +  _conn->escapeString(qMsg.description) + ", " + to_string(qMsg.timestamp) + ")";
```

Reply:
I found breaking it up easier to read, and to ensure escaped quotes matched.

Reply:
That's fine. Please, replace `boost::lexical_cast<string>` with `std::to_string`.